### PR TITLE
add conversion from and to v1alpha2 using Templates..Resources

### PR DESCRIPTION
**Pull request by anoipm**
**Description**

Add conversion from and to v1alpha2 using Templates..Resources

**Related issue(s)**
See also #15737 (this PR is only part of this issue)

### CODE REVIEW

File: components/function-controller/internal/webhook/conversion_webhook.go [hunk 1]
```diff
@@ -216,10 +218,14 @@ func convertResourcesV1Alpha1ToV1Alpha2(in *serverlessv1alpha1.Function, out *se
 	}
 
 	if buildResourcesExists {
+		//TODO: remove ResourceConfiguration in next step
 		out.Spec.ResourceConfiguration.Build.Resources = &in.Spec.BuildResources
+		setV1Alpha2BuildJobResources(in.Spec.BuildResources, out)
 	}
 	if functionResourcesExists {
+		//TODO: remove ResourceConfiguration in next step
```
Comment (pPrecel) [inline components/function-controller/internal/webhook/conversion_webhook.go:226]: Is this PR ready to review? 🤔

Comment (anoipm) [reply components/function-controller/internal/webhook/conversion_webhook.go:226]: Yes, this part will be removed after changes in controller which are part of the same issue.

**Review by pPrecel — APPROVED** [review]

**Pull request merged**